samples/interleave/sample.go

```diff
@@ -0,0 +1,803 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"flag"
+	"fmt"
+	"math/rand"
+	"os"
+	"strings"
+	"time"
+
+	spannergorm "github.com/googleapis/go-gorm"
+	_ "github.com/googleapis/go-sql-spanner"
+	"gorm.io/datatypes"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+type Singer struct {
+	gorm.Model
+	FirstName sql.NullString
+	LastName  string
+	// FullName is generated by the database. The '->' marks this a read-only field.
+	FullName string `gorm:"->;type:STRING(100) AS (ARRAY_TO_STRING([first_name, last_name], \" \")) STORED;"`
```

Comment:
Will `STRING(100)` be enough? How long could FirstName and LastName be?

Reply:
Changed to MAX